### PR TITLE
Discount orderbook retrieval from the solver time.

### PR DESCRIPTION
**Pull request by nlordell**
This PR implements discounting the orderbook retrieval time from the total time to solve for the solver. As the orderbook retrieval time grows, this can cause us to miss our deadline.

### Test Plan

New unit test.

### CODE REVIEW

**Review by e00E — APPROVED** [review]
I am fine with this too but it would be cleaner to pass in a deadline from the beginning instead of converting between duration and instant several times.

**Comment by nlordell** [comment]
> it would be cleaner to pass in a deadline from the beginning instead of converting between duration and instant several times.

Good point.

File: driver/src/driver/stablex_driver.rs [hunk 1]
```diff
@@ -143,12 +143,23 @@ impl<'a> StableXDriverImpl<'a> {
...
+        let deadline = Instant::now() + time_limit;
+
         self.metrics.auction_processing_started(&Ok(batch_to_solve));
         let (account_state, orders) = match self.get_orderbook(batch_to_solve) {
             Ok(ok) => ok,
             Err(err) => return DriverResult::Retry(err),
         };
-        match self.solve(batch_to_solve, time_limit, account_state, orders) {
+
+        let actual_time_limit = match deadline.checked_duration_since(Instant::now()) {
```
Comment (fleupold) [inline driver/src/driver/stablex_driver.rs]: nit: maybe rename to price_finding_time_limit / remaining_time_limit?

**Review by fleupold — APPROVED** [review]
🍠

**Pull request merged**